### PR TITLE
Cleanup upcoming paras

### LINKED ISSUE #1511: Ensure `schedule_para_cleanup` on Paras removes pending
If `schedule_para_initialize` and `schedule_para_cleanup` are called on the `Paras` module before a session boundary, the para will be initialized at the beginning of the next session. This behavior is not intended and undesirable.

### PR DESCRIPTION
**Pull request by montekki**
Closes #1511 

Decided to steal this one also while I'm at it

### CODE REVIEW

File: runtime/parachains/src/paras.rs [hunk 1]
```diff
@@ -412,15 +412,28 @@ impl<T: Trait> Module<T> {
 	/// Schedule a para to be cleaned up at the start of the next session.
 	#[allow(unused)]
 	pub(crate) fn schedule_para_cleanup(id: ParaId) -> Weight {
-		OutgoingParas::mutate(|v| {
+		let upcoming_weight = UpcomingParas::mutate(|v| {
+			match v.binary_search(&id) {
+				Ok(i) => {
+					v.remove(i);
```
Comment (rphmeier) [inline runtime/parachains/src/paras.rs:418]: Should remove from `UpcomingParasGenesis` as well, right?

File: runtime/parachains/src/paras.rs [hunk 2]
```diff
@@ -1150,6 +1163,69 @@ mod tests {
...
+			assert_eq!(<Paras as Store>::UpcomingParas::get(), vec![c, b, a]);
+			assert!(<Paras as Store>::Parathreads::get(&a).is_none());
+
+			Paras::schedule_para_cleanup(c);
+
+			run_to_block(3, Some(vec![3]));
+
+			assert_eq!(Paras::parachains(), vec![b]);
+			assert_eq!(<Paras as Store>::OutgoingParas::get(), vec![]);
+			assert_eq!(<Paras as Store>::UpcomingParas::get(), Vec::new());
```
Comment (rphmeier) [inline runtime/parachains/src/paras.rs:1219]: Test that upcoming-genesis is gone too

File: runtime/parachains/src/paras.rs (hunk 1 above)
Comment (montekki) [reply runtime/parachains/src/paras.rs:418]: true, thanks

**Comment by rphmeier** [comment]
bot merge

**Pull request merged**